src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -5080,6 +5251,62 @@ SpinelNCPInstance::remove_multicast_address_on_ncp(const struct in6_addr &addr,
 	start_new_task(factory.finish());
 }
 
+void
+SpinelNCPInstance::add_service_on_ncp(uint32_t enterprise_number, 
+	const uint8_t *service_data, unsigned int service_data_len, bool stable, 
+	const uint8_t *server_data, unsigned int server_data_len, CallbackWithStatus cb)
+{
+	SpinelNCPTaskSendCommand::Factory factory(this);
+
+	syslog(LOG_NOTICE, "Adding service with enterprise number:%u to NCP", enterprise_number);
```

Comment:
Suggestion: how about checking the capability for `SPINEL_CAP_THREAD_SERVICE` and if it's not supported, invoking callback `cb` with error?

Reply:
Done